src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseDataFileServiceTests.cs

```diff
@@ -1644,7 +1644,144 @@ public async Task ListAll()
+                    Filename = "test-data-2.csv",
+                    ContentLength = 20480,
+                    Type = FileType.Data,
+                    Created = DateTime.UtcNow,
+                    CreatedById = _user.Id,
+                    ReplacingId = originalFileId
+                }
+            };
+            var amendmentReplacementMetaReleaseFile = new ReleaseFile
+            {
+                ReleaseVersion = amendmentReleaseVersion,
+                File = new File
+                {
+                    Filename = "test-data-2.meta.csv",
+                    Type = Metadata,
+                }
+            };
+
+            var dataImports = new List<DataImport>
+            {
+                new()
+                {
+                    File = originalReleaseFile.File,
+                    MetaFile = originalMetaReleaseFile.File,
+                    TotalRows = 200,
+                    Status = COMPLETE
+                },
+                new()
+                {
+                    File = amendmentReplacementReleaseFile.File,
+                    MetaFile = amendmentReplacementMetaReleaseFile.File,
+                    TotalRows = 400,
+                    Status = COMPLETE
+                }
+            };
 
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                contentDbContext.ReleaseVersions.AddRange(releaseVersion, amendmentReleaseVersion);
+                contentDbContext.ReleaseFiles.AddRange(
+                    originalReleaseFile,
+                    originalMetaReleaseFile,
+                    amendmentOriginalReleaseFile,
+                    amendmentOriginalMetaReleaseFile,
+                    amendmentReplacementReleaseFile,
+                    amendmentReplacementMetaReleaseFile);
+                contentDbContext.DataImports.AddRange(dataImports);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var service = SetupReleaseDataFileService(contentDbContext: contentDbContext);
+
+                var result = await service.ListAll(releaseVersion.Id);
+
+                Assert.True(result.IsRight);
+
+                var files = result.Right.ToList();
+                
+                Assert.Single(files);
+
+                Assert.Equal("Test subject 1", files[0].Name);
+                Assert.Equal("test-data-1.csv", files[0].FileName);
+                Assert.Equal("csv", files[0].Extension);
+                Assert.Equal("test-data-1.meta.csv", files[0].MetaFileName);
+                Assert.Equal(_user.Email, files[0].UserName);
+                Assert.Equal(200, files[0].Rows);
+                Assert.Equal("10 Kb", files[0].Size);
+                Assert.Equal(COMPLETE, files[0].Status);
```

Comment:
I think you could remove these assertions as they aren't that important to this variant of the `ListAll` test and make it less obvious which is the important assertion that we want to check here. Maybe just leave one other  check in, like asserting on `Id` or the filename.